src/platform_impl/web/cursor.rs

```diff
@@ -163,6 +128,26 @@ impl CustomCursor {
     }
 }
 
+impl Hash for CustomCursor {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        Arc::as_ptr(&self.state).hash(state);
+    }
+}
+
+impl PartialEq for CustomCursor {
+    fn eq(&self, other: &Self) -> bool {
+        Arc::ptr_eq(&self.state, &other.state)
+    }
+}
+
+impl Eq for CustomCursor {}
```

Comment:
Nit: Keep these impls above, to reduce diff noise

Reply:
I don't think it matters here. Generally the order is `def -> impl def -> impl foo for def`, moving it doesn't really matter here, it's not like the diff is really big.